Source/NSubstitute/Core/PendingSpecification.cs

```diff
@@ -1,29 +1,65 @@
+using System;
+
 namespace NSubstitute.Core
 {
     public class PendingSpecification : IPendingSpecification
     {
-        ICallSpecification _pendingSpec;
+        private readonly ISubstitutionContext _substitutionContext;
+
+        public PendingSpecification(ISubstitutionContext substitutionContext)
+        {
+            _substitutionContext = substitutionContext;
+        }
+
+        public bool HasPendingCallSpecInfo()
+        {
+            //Check ownership to avoid potential issues when spec from foreign substitution is present.
+            var currentInfo = _substitutionContext.ThreadLocalPendingSpecificationInfo as PendingSpecificationInfo;
+            return currentInfo != null && currentInfo.Owner == this;
+        }
 
-        public bool HasPendingCallSpec()
+        public IPendingSpecificationInfo UseCallSpecInfo()
         {
-            return _pendingSpec != null;
+            var info = _substitutionContext.ThreadLocalPendingSpecificationInfo as PendingSpecificationInfo;
```

Comment:
Is there a reason to cast as `PendingSpecificationInfo` rather than lifting the required members to the interface?